Add documentation and use flag for CELER_USE_PROFILING (#1966)
celeritas-project/celeritas: example/offload-template/README.rst

  master and worker threads, setting up a tracking manager under the hood.
  :cpp:class:`EndOfRunAction` clears data and finalizes Celeritas data.

:cpp:class:`EventAction`
  The event action is not necessary to run Celeritas, but its use of
  :cpp:class:`celeritas::ScopedProfiling` demonstrates how to profile Geant4

>   and annotate GPU profiling results in a Geant4 context.

Reviewer: Why the focus on 'not being necessary' since those are listed under "Example classes"?
```suggestion
  The event action use of
  :cpp:class:`celeritas::ScopedProfiling` demonstrates how to profile Geant4
  and annotate GPU profiling results in a Geant4 context.
```

Author: Fair enough. I was trying to contrast it with the "minimal necessary" aspect of this example.

Reviewer: ```suggestion
  The event action is optional and its use of
  :cpp:class:`celeritas::ScopedProfiling` demonstrates how to profile Geant4
  and annotate GPU profiling results in a Geant4 context.
```
